securedrop/source_app/__init__.py

```diff
@@ -0,0 +1,118 @@
+from flask import (Flask, render_template, flash, Markup, request, g, session,
+                   abort, url_for, redirect)
+from flask_babel import gettext
+from flask_assets import Environment
+from flask_wtf.csrf import CSRFProtect
+from jinja2 import evalcontextfilter
+from os import path
+from sqlalchemy.orm.exc import MultipleResultsFound, NoResultFound
+
+import crypto_util
+import i18n
+import store
+import template_filters
+import version
+
+from db import Source, db_session
+from request_that_secures_file_uploads import RequestThatSecuresFileUploads
+from source_app import main, info, api
+from source_app.decorators import ignore_static
+from source_app.utils import logged_in
+
+
+def create_app(config):
+    app = Flask(__name__,
+                template_folder=config.SOURCE_TEMPLATES_DIR,
+                static_folder=path.join(config.SECUREDROP_ROOT, 'static'))
+    app.request_class = RequestThatSecuresFileUploads
+    app.config.from_object(config.SourceInterfaceFlaskConfig)
+
+    # The default CSRF token expiration is 1 hour. Since large uploads can
+    # take longer than an hour over Tor, we increase the valid window to 24h.
+    app.config['WTF_CSRF_TIME_LIMIT'] = 60 * 60 * 24
+    CSRFProtect(app)
+
+    assets = Environment(app)
+    app.config['assets'] = assets
+
+    # this set needs to happen *before* we set the jinja filters otherwise
+    # we get name collisions
```

Comment:
OK we discussed this in chat and decided that in a followup PR we are going to change the name of `datetimeformat` such that these collisions do not exist in order to avoid confusion. @heartsucker made issue #2374 for this and I've added it to the 0.4.4 milestone. Since the collision was not introduced in this PR, this issue is non-blocking, but the name collision needs to be fixed prior to releasing these changes in 0.4.4.